src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -488,6 +488,16 @@ private BoundForEachStatement BindForEachPartsWorker(DiagnosticBag diagnostics,
                 hasErrors |= GetAwaitDisposeAsyncInfo(ref builder, diagnostics);
             }
 
+            Debug.Assert(
+                hasErrors ||
+                builder.CollectionConversion.IsIdentity ||
+                (builder.CollectionConversion.IsImplicit &&
+                 (IsIEnumerable(builder.CollectionType) ||
+                  IsIEnumerableT(builder.CollectionType.OriginalDefinition, IsAsync, Compilation))) ||
+                // To compat behavior, we can enumerate over System.String even if it's not IEnumerable. That will
```

Comment:
>String [](start = 68, length = 6)

Is there a similar treatment for Span (which we allow enumerating over, but I don't think implements IEnumerable)?